src/plugins/intel_npu/src/compiler/include/zero_compiler_in_driver.hpp

```diff
@@ -67,70 +69,38 @@ class LevelZeroCompilerInDriver final : public IExternalCompiler {
+     * --inputs_precisions="0:<input1Precision> [1:<input2Precision>]"
+     * --inputs_layouts="0:<input1Layout> [1:<input2Layout>]"
+     * --outputs_precisions="0:<output1Precision>"
+     * --outputs_layouts="0:<output1Layout>"
+     *
+     * For older compiler versions, the name of the inputs/outputs may be used instead of their indices.
      *
      * Since the layout information is no longer an important part of the metadata values when using the 2.0 OV
      * API, the layout fields shall be filled with default values in order to assure the backward compatibility
      * with the driver.
      */
-    static std::string serializeIOInfo(const std::shared_ptr<const ov::Model>& model);
+    static std::string serializeIOInfo(const std::shared_ptr<const ov::Model>& model, const bool useIndices);
 
 private:
     NetworkMetadata getNetworkMeta(ze_graph_handle_t graphHandle) const;
 
     std::vector<uint8_t> serializeIR(IR& irModel, ze_graph_compiler_version_info_t compilerVersion) const;
     std::string serializeConfig(const Config& config, ze_graph_compiler_version_info_t& compilerVersion) const;
 
-    /**
-     * @brief Extracts the layout value or the state descriptor from the given Level Zero structure.
-     * @details Extracting the layout information is required only when using older driver versions which rely on
-     * this legacy attribute. Since this information is not found within the parameter/result nodes, we need to
-     * extract this value here.
-     *
-     * The state variables are also not found in the previously mentioned nodes, thus if the given Level Zero
-     * parameter corresponds to an input/output, we shall extract the layout value from it. Else it represents a
-     * state variable and the descriptor will be extracted and stored in an OpenVINO specific format.
-     * @param parameters Holds the already extracted input node descriptors. The transposed shape attribute of the
-     * corresponding entry may be updated according to the extracted layout value.
-     * @param results Holds the already extracted output node descriptors. The transposed shape attribute of the
-     * corresponding entry may be updated according to the extracted layout value.
-     * @param states The state descriptors shall be stored here in an OpenVINO specific format.
-     * @param stateNames The output location of the state variables' names in the order found within the compiled
-     * model.
-     * @param arg The Level Zero specific structure from which the layout value or state variable descriptor shall
-     * be extracted.
-     */
-    template <typename T>
-    void getLayoutOrStateDescriptor(IONodeDescriptorMap& parameters,
-                                    IONodeDescriptorMap& results,
-                                    IONodeDescriptorMap& states,
-                                    std::vector<std::string>& stateNames,
-                                    const T& arg) const;
```

Comment:
We're not sending layouts anymore (other than the default one), so it's safe to assume extracting the layout is now a redundant operation.